homeassistant/components/unifiprotect/entity.py

```diff
@@ -203,3 +224,106 @@ def _async_update_device_from_protect(self) -> None:
             self.device = self.data.api.bootstrap.nvr
 
         self._attr_available = self.data.last_update_success
+
+
+class AccessTokenMixin(Entity):
+    """Adds access_token attribute and provides access tokens for use for anonymous views."""
+
+    @property
+    def access_tokens(self) -> deque[str]:
+        """Get valid access_tokens for current entity."""
+        assert isinstance(self, ProtectDeviceEntity)
+        return self.data.async_get_or_create_access_tokens(self.entity_id)
+
+    @callback
+    def _async_update_and_write_token(self) -> None:
+        _LOGGER.debug("Updating access tokens for %s", self.entity_id)
+        self.async_update_token()
+        self.async_write_ha_state()
+
+    @callback
+    def async_update_token(self) -> None:
+        """Update the used token."""
+        self.access_tokens.append(
+            hashlib.sha256(_RND.getrandbits(256).to_bytes(32, "little")).hexdigest()
+        )
+
+    @callback
+    def _trigger_update_tokens(self, *args: Any, **kwargs: Any) -> None:
+        assert isinstance(self, ProtectDeviceEntity)
+        async_dispatcher_send(
+            self.hass,
+            f"{EVENT_UPDATE_TOKENS}-{self.entity_description.key}-{self.entity_id}",
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added to hass.
+
+        Injects callbacks to update access tokens automatically
+        """
+        await super().async_added_to_hass()
+
+        self.async_update_token()
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                f"{EVENT_UPDATE_TOKENS}-{self.entity_id}",
+                self._async_update_and_write_token,
+            )
+        )
+        self.async_on_remove(
+            self.hass.helpers.event.async_track_time_interval(
+                self._trigger_update_tokens, TOKEN_CHANGE_INTERVAL
+            )
+        )
+
```

Comment:
Do we need to cleanup anything in `self.data` if the entity is renamed/disabled ?

If so we should probably do something in `async_will_remove_from_hass`

Reply:
Probably a good idea. If the entity no longer exists for some reason, the view should also likewise fail.